src/main/java/iudx/aaa/server/registration/RegistrationServiceImpl.java

```diff
@@ -513,15 +472,17 @@ public RegistrationService updateUser(UpdateProfileRequest request, User user,
         response.put(RESP_ORG, details);
       }
 
-      LOGGER.info("Updated user profile for " + u.getUserId().toString() + " with roles "
-          + request.getRoles().toString());
+      LOGGER.info("Updated user profile for " + u.getUserId().toString() + " (" + title + ")");
 
-      Response r = new ResponseBuilder().type(URN_SUCCESS).title(SUCC_TITLE_UPDATED_USER_ROLES)
-          .status(200).objectResults(response).build();
+      Response r = new ResponseBuilder().type(URN_SUCCESS).title(title).status(200)
+          .objectResults(response).build();
       handler.handle(Future.succeededFuture(r.toJson()));
     }).onFailure(e -> {
-      if (e.getMessage().equals(COMPOSE_FAILURE)) {
-        return; // do nothing
+
+      if (e instanceof ComposeException) {
+        ComposeException exp = (ComposeException) e;
+        handler.handle(Future.succeededFuture(exp.getResponse().toJson()));
+        return;
       }
       LOGGER.error(e.getMessage());
       handler.handle(Future.failedFuture("Internal error"));
```

Comment:
Since this is in the end of the service, if there were any compose failures, either from the methods or from the service itself, we can handle it here and send back the JSON response